app/main.py

```diff
@@ -1 +1,22 @@
-# write your code here
+from typing import List
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: List[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    any_vaccine_error = False  # Add a flag
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            any_vaccine_error = True  # Set flag on VaccineError
```

Comment:
The current logic sets `any_vaccine_error` to `True` and continues the loop. However, ensure that this logic does not prematurely terminate the function, preventing the counting of masks needed. Consider restructuring the logic to ensure all friends are processed before returning a result.